gulpfile.js

```diff
@@ -40,19 +41,41 @@ gulp.task('ext:lint', () => {
     .pipe(gulpTsLint.report());
 });
 
+// Copy icons for OE
 gulp.task('ext:copy-assets', (done) => {
     return gulp.src([
         config.paths.project.root + '/src/objectExplorer/objectTypes/*'
     ])
     .pipe(gulp.dest('out/src/objectExplorer/objectTypes'));
 });
 
+// Compile source
 gulp.task('ext:compile-src', (done) => {
     return gulp.src([
-        config.paths.project.root + '/src/**/*.ts',
-        config.paths.project.root + '/src/**/*.js',
-        config.paths.project.root + '/typings/**/*.ts',
-        '!' + config.paths.project.root + '/src/views/htmlcontent/**/*'])
+                config.paths.project.root + '/src/**/*.ts',
+                config.paths.project.root + '/src/**/*.js',
+                '!' + config.paths.project.root + '/typings/**/*.ts',
+                '!' + config.paths.project.root + '/src/views/htmlcontent/**/*'])
+                .pipe(srcmap.init())
+                .pipe(tsProject())
+                .on('error', function() {
+                    if (process.env.BUILDMACHINE) {
+                        done('Extension Tests failed to build. See Above.');
+                        process.exit(1);
+                    }
+                })
+                .pipe(nls.rewriteLocalizeCalls())
+                .pipe(nls.createAdditionalLanguageFiles(nls.coreLanguages, config.paths.project.root + '/localization/i18n', undefined, false))
+                .pipe(srcmap.write('.', {
+                   sourceRoot: function(file){ return file.cwd + '/src'; }
+                }))
+                .pipe(gulp.dest('out/src/'));
+});
+
+// Compile view source
+gulp.task('ext:compile-view', (done) => {
```

Comment:
for some reason I was under the impression we were removing the the separate build steps for the "view" code and just have one normal build?